felt/src/lib.rs

```diff
@@ -877,6 +895,43 @@ mod test {
             prop_assert!(&x.to_biguint() < p);
         }
 
+        #[test]
+        #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+        fn to_be_bytes(ref x in FELT_PATTERN) {
+            let x = &Felt252::from_bytes_be(x.as_bytes());
+            let bytes = x.to_be_bytes();
+            let y = &Felt252::from_bytes_be(&bytes);
+            prop_assert!(x == y);
+        }
+
+        #[test]
+        #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+        fn to_le_bytes(ref x in FELT_PATTERN) {
+            let x = &Felt252::from_bytes_be(x.as_bytes());
+            let mut bytes = x.to_le_bytes();
+            // Convert to big endian for test
+            bytes.reverse();
+            let y = &Felt252::from_bytes_be(&bytes);
+            prop_assert!(x == y);
+        }
+
+        #[test]
+        #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+        fn to_u128_ok(ref x in "(0|[1-9a-f][0-9a-f]{0,31})") {
+            let x = &Felt252::parse_bytes(x.as_bytes(), 16).unwrap();
+            let y = x.to_u128();
+            prop_assert!(y.is_some());
+            prop_assert!(x.to_string() == y.unwrap().to_string());
+        }
+
+        #[test]
+        #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+        fn to_u128_out_of_range(ref x in "([1-9a-f][0-9a-f]{32,})") {
```

Comment:
Limit the range to make sure it doesn't wrap around the prime and ends up in range for `u128`:
```suggestion
        fn to_u128_out_of_range(ref x in "([1-9a-f][0-9a-f]{32,63})") {
```